README.rst

```diff
@@ -97,6 +97,14 @@ For instance, ``has_attrs(name="bob")`` is equivalent to ``has_attrs(name=equal_
       # Matches ["a", "b"], ["b", "a"] and ["a", "c", "b"]
       # but not ["a", "c"] nor ["a"]
 
+* ``contains_only(matcher)``: matches an iterable if every element matches `matcher`.
```

Comment:
This name feels ambiguous to me, since it could also mean the same thing as, say, `contains_exactly` (perhaps with different behaviour with repeated elements).

Perhaps the name `all_elements` would be clearer? e.g. `all_elements(equal_to(42))`.

Reply:
`contains_only` -> `all_elements`